### PR TITLE
Fixes #7050: Add a rudder server disable command

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7050

### CODE REVIEW

**Comment by Kegeruneku** [comment]
I'm pretty sure that stopping the agent during disable should be the default, I (at least) would expect if I "disable the agent" that it it turned off completely afterwards :) (POLA and all)

Maybe you should reverse the logic and change the switch to "do not turn the agent off" ?

**Comment by peckpeck** [comment]
If so it should not be named disable but stop

**Comment by jooooooon** [comment]
I don't think that these 2 different services should be mixed together. I see plenty of valid reasons to want to stop just the policy distribution (ie, "server stop" or maybe "server policy stop" since we have several servers in effet), such as "I want to continue enforcing current configs but not roll out any changes right now". In such a case, it would be desirable to have rudder-agent continue to run (just not get any policy updates).

The process to be managed is actually cf-serverd, not just rudder-agent. The init script for rudder-agent runs both cf-serverd and cf-execd (which in turn runs rudder-agent). So, IMHO, POLA would indicate that this command only stop cf-serverd, not cf-execd or rudder-agent.

**Comment by peckpeck** [comment]
But if we only stop cf-served rudder agent will rerun it as soon as it runs.

**Comment by jooooooon** [comment]
We could use a flag file on the FS that our policy checks before restarting cf-serverd? A bit like disable-rudder-agent, but disable-rudder-policy-server or similar?

**Comment by peckpeck** [comment]
couldn't we just merge the command and add an option later to only stop the server ?

File: share/commands/agent-disable [hunk 1]
```diff
@@ -2,5 +2,41 @@
...
+then
+  if [ -n "${CMD}" ]
+  then
+    $CMD
+  else
+    echo "Don't know how to stop rudder agent." 1>&2
+    echo "Agent not stopped !" 1>&2
+  fi
+else
+  echo "rudder-agent has been enabled but not started, wait for next cron run"
```
Comment (jooooooon) [inline share/commands/agent-disable]: s/enabled/disabled/ + s/started/stopped/ + remove the bit about a cron run

Comment (jooooooon) [inline share/commands/agent-disable]: I propose to add "run rudder agent disable -s to stop the agent" too

**Comment by jooooooon** [comment]
I'm fine with the "rudder agent disable" command as is (minus the backwards warning message I commented on above), but I'm really not comfortable having a "rudder server disable" that only disables one small part of the server, and even then, as Matthieu says, won't ever actually *stop* it by default.

**Comment by peckpeck** [comment]
Disabling only the promise distribution is a real use case we met. We want the server to continue running, but we don't trust what's in tit to be used by the agents.

May be we can add a mandatory parameter --cfengine which we'll expand later with others like --webapp, --apache ... and a --all

**Comment by jooooooon** [comment]
That sounds better. Not "--cfengine" though, how about "--policy-server"?

**Comment by peckpeck** [comment]
PR updated

**Comment by jooooooon** [comment]
We agreed on naming the command "rudder server disable-policy-distribution" and adding a "flag file" /opt/rudder/etc/disable-policy-distribution so that the agent doesn't start cf-serverd again.

**Comment by jooooooon** [comment]
Replaced by #62

**Pull request closed without merging**